slack/web/slack_response.py

```diff
@@ -132,13 +135,27 @@ def __next__(self):
                 {"cursor": self.data["response_metadata"]["next_cursor"]}
             )
 
-            response = asyncio.get_event_loop().run_until_complete(
-                self._client._request(
-                    http_verb=self.http_verb,
-                    api_url=self.api_url,
-                    req_args=self.req_args,
+            if self._use_sync_aiohttp:
+                # We no longer recommend going with this way
+                response = asyncio.get_event_loop().run_until_complete(
+                    self._client._request(
+                        http_verb=self.http_verb,
+                        api_url=self.api_url,
+                        req_args=self.req_args,
+                    )
                 )
-            )
+            else:
+                if self._client is None:
+                    # NOTE: It's not possible to support this due to circular import issues.
+                    msg = (
+                        "Directly using UrllibWebClient doesn't support response pagination as iteration. "
```

Comment:
Is `UrllibWebClient` meant to be a part of the public API at all (regardless of using pagination with an iterator)? If not, then I think this message is a little too specific. Can we instead fail in the constructor when `_use_sync_aiohttp=False` and there's no `client`?

Reply:
This is a good point. I was also wondering if it's really necessary to support directly the case using `UrllibWebClient`. Failing in a constructor would be easier to understand.

Reply:
This warning message has been removed. After replying as above, I came to the conclusion we should not raise errors in the constructor. The reason is that although the pagination interactor with `use_sync_aiohttp=True` is not efficient, it still works.
see also: https://github.com/slackapi/python-slackclient/pull/662/commits/80dbec7ba065f8b00c3e37794996e232b9c340fd